static/report/js/compare.js

```diff
@@ -0,0 +1,639 @@
+    var pool = pools.filter(function (pool) {
+      var timeline = pool.reticle.dom.find('.timeline');
+      return timeline.is(target);
+    }).pop( );
+    if (pool) {
+      pool.display.control.setRange(range[0], range[1]);
+    }
+  }
+
+  function add_new ( ) {
+    var item = make({ });
+    pools.push(item);
+  }
+
+  manage.add_new = add_new;
+
+  return init( );
+}
+
+function pager (opts) {
+  var query = { };
+  var url = opts.url || '/api/v1/entries.json?find[type]=sgv&find[sgv][$gt]=39&count=500000&';
+  query.begin = opts.begin;
+  query.end = opts.end;
+  var payload = [ ];
+  var days = { };
+  var cache = opts.cache;
+  var DATE_FMT = "{yyyy}-{MM}-{dd}";
+
+  var bisect = d3.bisector(function (d) { return Date.create(d.dateString); });
+  function page ( ) {
+  }
+
+  function refresh (start, end) {
+    // console.log("PAYLOAD?", payload);
+    var q = {
+      start: Date.create(Date.create(start).format(DATE_FMT))
+    , end: Date.create(Date.create(end).format(DATE_FMT))
+    };
+    query.begin = start;
+    query.end = end;
+    var range = d3.time.days(q.start, q.end);
+    days = { };
+    console.log('QUERY FOR', range.length, 'days', q, query);
+    iter_query(range);
+    return;
+    if (payload && payload.length > 0) {
+      console.log("NUM DAYS", range.length);
+      // TODO: soft update, only get deltas against the edges of the
+      // cursor.
+      // start.isBefore
+      // start.isBetween
+      // start.isAfter
+      // end.isBefore
+      // end.isBetween
+      // end.isAfter
+      var holding = {
+          begin: Date.create(payload[0].dateString)
+        , last:  Date.create(payload.slice(-1).pop( ).dateString)
+      }
+      console.log("BISECT LEFT start", bisect.left(payload, { dateString: start }));
+      console.log("BISECT LEFT end", end, bisect.right(payload, { dateString: end }));
+      do_query(start, end, first_page);
+    } else {
+      do_query(start, end, first_page);
+    }
+    query.begin = start;
+    query.end = end;
+  }
+
+  function param_string (begin, end) {
+    return [
+      "find[dateString][$gte]=" + Date.create(begin).format(DATE_FMT)
+    , "find[dateString][$lte]=" + Date.create(end).format(DATE_FMT)
+    ].join('&')
+  }
+
+  function iter (prev, current, index, tail) {
+  }
+
+  function iter_query (range) {
+    payload = [ ];
+    range.forEach(function (day, i) {
+      var start = day;
+      var end = d3.time.day.offset(start, 1);
+      var data = cache.within(start, end);
+      if (data.length < 10) {
+        do_query(start, end, function _iter_ (resp) {
+          days[Date.create(day).format(DATE_FMT)] = true;
+          collate(resp);
+        });
+      } else {
+        days[Date.create(day).format(DATE_FMT)] = true;
+        payload = payload.concat(data);
+        if (Object.keys(days).length >= range.length) {
+          do_payload( );
+        }
+      }
+    });
+
+  }
+
+  function accrue (begin, end, cb) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: ["accrue" is defined but never used](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2715308466/issues/source?bid=2452374&fileBranchId=3345581#l340)